Machine_Learning/Supervised_Machine_Learning/ML_2811_K-Nearest-Neighbour-Algorithm_(D).md

```diff
@@ -0,0 +1,167 @@
+![image](https://user-images.githubusercontent.com/71644914/135436797-bc04081b-d845-44ab-8728-639aab394187.png)
+
+
+
+Suppose P1 is the point, for which label needs to predict. First, you find the k closest point to P1 and then classify points by majority vote of its k neighbours. Each object votes for their class and the class with the most votes is taken as the prediction. For finding closest similar points, you find the distance between points using distance measures such as Euclidean distance, Hamming distance, Manhattan distance and Minkowski distance. KNN has the following basic steps:
+1.	Calculate distance
+2.	Find closest neighbours
+3.	Vote for labels
+
+![image](https://user-images.githubusercontent.com/71644914/135437123-19be0137-528c-4bda-95ba-247ea9bf6b12.png)
+
+
+**How do you decide the number of neighbours in KNN?**
+
+Now, you understand the KNN algorithm working mechanism. At this point, the question arises that How to choose the optimal number of neighbours? And what are its effects on the classifier? The number of neighbours(K) in KNN is a hyperparameter that you need choose at the time of model building. You can think of K as a controlling variable for the prediction model.
+Research has shown that no optimal number of neighbours suits all kind of data sets. Each dataset has it's own requirements. In the case of a small number of neighbours, the noise will have a higher influence on the result, and a large number of neighbours make it computationally expensive. Research has also shown that a small amount of neighbours are most flexible fit which will have low bias but high variance and a large number of neighbours will have a smoother decision boundary which means lower variance but higher bias.
+Generally, Data scientists choose as an odd number if the number of classes is even. You can also check by generating the model on different values of k and check their performance. You can also try Elbow method here.
+
+![image](https://user-images.githubusercontent.com/71644914/135437232-db74a35a-7103-43d0-9df8-9e5a44ba688b.png)
+
+
+# Implementing KNN Algorithm with Scikit-Learn
+
+**The Dataset:-**
+We are going to use the famous iris data set for our KNN example. The dataset consists of four attributes: sepal-width, sepal-length, petal-width and petal-length. These are the attributes of specific types of iris plant. The task is to predict the class to which these plants belong. There are three classes in the dataset: Iris-setosa, Iris-versicolor and Iris-virginica.
+
+**Importing Libraries:-**
+
+import numpy as np
```

Comment:
The codes you have been written are not being distinguished, it looks like normal write up. You can add up the codes like 
``` python
import numpy as np
import matplotlib as plt
import pandas as pd
```

@monalsingh10

Reply:
Also the file name should not be like this. It should be like, 
`ML_2_8_K-Nearest Neighbour Algorithm_(D).md`

Reply:
Okay sir I will be doing it.

Reply:
@abhisheks008 Sir in document I have added libraries that is pandas , matplotlib , numpy  with proper code.

Reply:
Yeah I know that, but it is not distinguished well enough. Kindly look into that.

Reply:
The code snippets will be written properly.

Reply:
Okay Sir!